spec/mailer/planning_application_mailer_spec.rb

```diff
@@ -59,7 +59,8 @@
   end
 
   before do
-    ENV["APPLICANTS_APP_HOST"] = "example.com"
+    allow(ENV).to receive(:[])
```

Comment:
Do we need this line as well, is this an extra step we need on top of the line below?

I would have thought just `allow(ENV).to receive(:[]).with("APPLICANTS_APP_HOST").and_return("example.com")` would suffice?

Reply:
Ah nvm i tested locally, that doesn't work..

I guess there aren't many cases we stub the env here but might be nice moving to a module under `spec/support` i.e.

```ruby
module EnvHelper
  def set_env(name:, value:)
    allow(ENV).to receive(:[])
    allow(ENV).to receive(:[]).with(name).and_return(value)
  end
end

if RSpec.respond_to?(:configure)
  RSpec.configure do |config|
    config.include(EnvHelper)
  end
end
```

then we can just do something like `set_env(name: "APPLICANTS_APP_HOST", value: "example.com")`

But not super fussed, will leave up to you

Reply:
The comment for the stackoverflow was:

`allow(ENV).to receive(:[])  # stub a default value first if message might be received with other args as well.`

When I tested the code without it and it blows up as other messages are indeed calling it:

- For the mailer it calls ENV with: "FARADAY_NO_AUTOLOAD"
- For the planning_application model spec:  "https_proxy"

I think your helper code looks cool; not seen `RSpec.resond_to?(:configure)`! I wanted to just get the fix it. There's an issue about ENV vs ENV.fetch. Some of the code uses one and some the other. 

Thoughtbot had this to say: 
`Use ENV.fetch for environment variables instead of ENV[]so that unset environment variables are detected on deploy.`

[Rubocop has introduced a cop for it.](https://msp-greg.github.io/rubocop/RuboCop/Cop/Style/FetchEnvVar.html)

It might make sense in a later commit to change ENV => ENV.fetch throughout the code and introduce a helper like you suggested.. fix it all in 1 sweep?

Reply:
Cool thanks for the explanation.
Yeah it's best to be consistent I suppose, might be worth adding that rubocop rule 👍 

Yep, will leave it up to you in terms of doing it as another commit / PR.